### PR TITLE
Modify after/before each hooks to take a result / spec parameter

### PR DESCRIPTION
**Pull request by stbenjam**
Allows a beforeEach to know the context it's operating in (but not modify it).

Allows an afterEach to modify a test result, collect diagnostics, etc.

Also aligns details to the enhancement.

### CODE REVIEW

File: pkg/extension/extensiontests/types.go [hunk 1]
```diff
@@ -18,13 +18,13 @@ type ExtensionTestSpec struct {
 
 	// OtherNames contains a list of historical names for this test. If the test gets renamed in the future,
 	// this slice must report all the previous names for this test to preserve history.
-	OtherNames sets.Set[string] `json:"otherNames"`
```
Comment (jupierce) [inline pkg/extension/extensiontests/types.go:21]: Enhancement currently relies on `originalName`, which cannot change. Every other name becomes, in effect, the current "display name". 
I'm not against storing the history, but I'm not confident there exists a performant "GROUP BY" mechanic to aggregate runs for all historical names using a list like this.

File: pkg/extension/extensiontests/result.go [hunk 2]
```diff
@@ -5,3 +5,7 @@ func (results ExtensionTestResults) Walk(walkFn func(*ExtensionTestResult)) {
 		walkFn(results[i])
 	}
 }
+
+func (result *ExtensionTestResult) AddDetails(name string, value interface{}) {
```
Comment (jupierce) [inline pkg/extension/extensiontests/result.go:9]: comment stating that value must be marshal into JSON?

File: pkg/extension/extensiontests/types.go (hunk 1 above)
Comment (stbenjam) [reply pkg/extension/extensiontests/types.go:21]: Let's discuss on the enhancement https://github.com/openshift/enhancements/pull/1676/files#r1799952825

**Pull request merged**